backend/helpers/helpers.go

```diff
@@ -72,9 +73,27 @@ func GetFirstTable[T models.AllTables, P models.AllTables](w http.ResponseWriter
 
 }
 
+func UpdateSpecifiedParams[T models.AllTables](w http.ResponseWriter, r *http.Request, table *T, initCommon models.Common, updatedCommon *models.Common) (error, T) {
+	err := json.NewDecoder(r.Body).Decode(&table)
+	if err != nil {
+		http.Error(w, "400 malformed request", http.StatusBadRequest)
+		return err, *table
+	}
+	if initCommon != *updatedCommon {
+		http.Error(w, "Invalid updates to automatic fields", http.StatusBadRequest)
+		return errors.New("400"), *table
+	}
```

Comment:
one is a pointer to the original object, which was updated and one isn't a pointer

Reply:
Every usage I see uses `foo` and `&foo`, the same variable before any modification. In what scenario could they be different?

Reply:
![image](https://github.com/lynx-locks/lynx-web/assets/57304476/8ea6aa7a-eead-4850-bea9-755d97e14edc)
in this scenario, cuz we do a search for backend by id, that goes into "foo", and then we pass by value foo.common, but also pass by reference foo.common, and pass by reference foo. In this case when we use the line below:
err := json.NewDecoder(r.Body).Decode(&table)

we are updating the table from the backend adding on top any values inputted by the user in the request body, this will also update the pass by reference foo.common but crucially not the pass by value one which retains the original values returned from the db. Now in the case of the image above, we would have a mismatch where the user added an id in the body which doesn't match the id of the object we are trying to update. (you can imagine why this would cause issues, esp cuz gorm will automatically insert if the id doesn't exist iirc). As such we do this comparison bass by reference foo doesn't equal pass by value food and we know there's been an invalid change to uneditable data.